src/library/scala/collection/immutable/LinkedMap.scala

```diff
@@ -0,0 +1,214 @@
+package scala
+package collection
+package immutable
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+
+import scala.collection.mutable.{Builder, ImmutableBuilder}
+
+/**
+  * A generic trait for ordered immutable maps. Concrete classes have to provide
+  * functionality for the abstract methods in `LinkedMap`.
+  *
+  * Note that when checking for equality [[LinkedMap]] does not take into account
+  * ordering.
+  *
+  * @tparam K      the type of the keys contained in this linked map.
+  * @tparam V      the type of the values associated with the keys in this linked map.
+  *
+  * @author Matthew de Detrich
+  * @version 2.13
+  * @since 2.13
+  * @define coll immutable linked map
+  * @define Coll `immutable.LinkedMap`
+  */
+
+trait LinkedMap[K, +V]
+  extends AbstractMap[K, V]
+    with MapOps[K, V, LinkedMap, LinkedMap[K, V]]
+
+object LinkedMap extends MapFactory[LinkedMap] {
+  def empty[K, V]: LinkedMap[K, V] = EmptyLinkedMap.asInstanceOf[LinkedMap[K, V]]
+
+  def from[K, V](it: collection.IterableOnce[(K, V)]): LinkedMap[K, V] =
+    it match {
+      case vm: LinkedMap[K, V] => vm
+      case _ => (newBuilder[K, V] ++= it).result()
+    }
+
+  def newBuilder[K, V]: Builder[(K, V), LinkedMap[K, V]] =
+    new MapFactory.Delegate[LinkedMap](VectorMap).newBuilder
```

Comment:
You can just call `VectorMap.newBuilder`.

Reply:
Cool, will add this tonight